tests/unit/models/test_config.py

```diff
@@ -14,18 +14,61 @@
 
 from __future__ import annotations
 
-import gpflow
-import numpy as np
 import pytest
-import tensorflow as tf
 
-from tests.util.models.gpflow.models import gpr_model
-from tests.util.models.models import fnc_3x_plus_10
-from trieste.models import ModelConfig
+from tests.util.models.gpflow.models import GPRcopy, gpr_copy_model
+from trieste.models import ModelConfig, ModelRegistry, create_model
+from trieste.models.gpflow import GaussianProcessRegression
+from trieste.models.optimizer import Optimizer
 
 
-def test_model_missing_abstract_method() -> None:
-    x = tf.constant(np.arange(5).reshape(-1, 1), dtype=gpflow.default_float())
-    model_specs = {"model": gpr_model(x, fnc_3x_plus_10(x))}
+def test_model_config_raises_not_supported_model_type() -> None:
+
+    model = gpr_copy_model()
     with pytest.raises(NotImplementedError):
-        ModelConfig(**model_specs)
+        ModelConfig(model)
+
+
+def test_model_registry_raises_on_unsupported_model() -> None:
+
+    model = gpr_copy_model()
+
+    with pytest.raises(ValueError):
+        ModelRegistry.get_interface(model)
+
+    with pytest.raises(ValueError):
+        ModelRegistry.get_optimizer(model)
+
+
+def test_model_registry_register_model() -> None:
+
+    ModelRegistry.register_model(GPRcopy, GaussianProcessRegression, Optimizer)
+    model_type = type(gpr_copy_model())
+
+    assert ModelRegistry.get_interface(model_type) == GaussianProcessRegression
+    assert ModelRegistry.get_optimizer(model_type) == Optimizer
+
+
+def test_model_registry_register_model_warning() -> None:
+
+    with pytest.warns(UserWarning) as record:
```

Comment:
It's generally bad to have tests that depend on other tests being run first. You can fix it here (and below) by explicitly (re)registering first. (It would be even better to ensure the global state is put back at the end of each of the tests, but that's harder without adding an unregister_model which probably isn't worth it.)